docs/charts/line.md

```diff
@@ -50,7 +50,7 @@ The line chart allows a number of properties to be specified for each dataset. T
 | [`borderDashOffset`](#line-styling) | `number` | Yes | - | `0.0`
 | [`borderJoinStyle`](#line-styling) | `string` | Yes | - | `'miter'`
 | [`borderWidth`](#line-styling) | `number` | Yes | - | `3`
-| [`cubicInterpolationMode`](#cubicinterpolationmode) | `string` | Yes | - | `''`
+| [`cubicInterpolationMode`](#cubicinterpolationmode) | `string` | Yes | - | `undefined`
```

Comment:
I think we should use the same default value for both `line.md` and `elements.md`

Reply:
@benmccann is right, other line options are also by default `undefined` but are documented with the element default values. We should maybe also update the "Point Styling" and "Line Styling" sections with:

> If the value is `undefined` (default), ...

to make a bit clearer that these dataset options fallback to the element options by default.

Reply:
@simonbrunel  @benmccann  in my opinion we should put the value that is set into the property if untouched (depends on code). If `undefined`, is correct to write undefined and leave the sentence written below

> If left untouched (`undefined`), the global `options.elements.line.cubicInterpolationMode` property is used.

Instead, if the property is set to `default`, even if untouched, we should write `default` as default and remove (or re-phrase) the sentence which describe the relations with options element.

Reply:
@simonbrunel @benmccann furthermore, I think `cubicInterpolationMode` can be set at global level but also at chart element line options level in order that all datasets of chart will use the same value, isn't it?

Reply:
The option value `'default'` is definitely not a good name (too generic and confusing). Anyway, all other line dataset options are also `undefined` by default but the docs promote the "effective" default. For example, `borderWidth` is undefined and fallbacks to the [element default](/chartjs/Chart.js/blob/master/src/elements/element.line.js#L16), while it's written in the doc:

```markdown
| [`borderWidth`](#line-styling) | `number` | Yes | - | `3`
```

So for consistency I would keep the same logic for `cubicInterpolationMode`:

```markdown
| [`cubicInterpolationMode`](#cubicinterpolationmode) | `string` | Yes | - | `'default'`
```

>  I think cubicInterpolationMode can be set at global level

I don't think so, where do you see that behavior in the code?

Reply:
Maybe I'm wrong but the hierarchy of configuration is:
```plain
Dataset options
         ^
Chart options
         ^
Global Chart options
         ^
Global options
```
therefore my expectation is that when a property can be set from lower level (if undefined and planned), the property can be set on all lower levels and not skipping one of them.
In our case, `cubicInterpolationMode` cna be set at dataset level and from 2.8.0 can be set also at global one but I think it should be set also at chart and global chart options.
Correct me if I'm saying something wrong.

Reply:
In 2.8.0 and for this specific option, [the hierarchy is](/chartjs/Chart.js/blob/master/src/controllers/controller.line.js#L211):

```
// fallback from top to bottom
- (instance) chart.data.datasets[i].cubicInterpolationMode 
- (instance) chart.options.elements.line.cubicInterpolationMode 
- (global) Chart.defaults.global.elements.line.cubicInterpolationMode 
```

Reply:
@simonbrunel ok, that's what I meant (I explained myself wrongly).
Coming back to the defualt value, I'll commit with "default" instead of "undefined"

Reply:
Thanks @stockiNail :) 

Actually, it should be `'default'` (as string) instead of `default`

Reply:
Done! 
@simonbrunel may I ask you if also `label` property for some datasets type is correct the default to `''`? Because I see that for others is `undefined`

Reply:
I really don't know what should be `label`

Reply:
> 
> 
> In 2.8.0 and for this specific option, [the hierarchy is](/chartjs/Chart.js/blob/master/src/controllers/controller.line.js#L211):
> 
> ```
> // fallback from top to bottom
> - (instance) chart.data.datasets[i].cubicInterpolationMode 
> - (instance) chart.options.elements.line.cubicInterpolationMode 
> - (global) Chart.defaults.global.elements.line.cubicInterpolationMode 
> ```

Maybe I'm wrong but in the fallback chain you described, the global chart settings are missing, settings which are stored in `Chart.defaults[chartType]`, for instance `Chart.defaults.line`. 
In this case, being a property only for chart line, nothing change.

Reply:
True, I forgot about `Chart.defaults[type].elements.line.cubicInterpolationMode` (I updated my previous comment).